File: src/test/java/no/nav/sosialhjelp/soknad/consumer/kodeverk/dto/KodeverkDtoTest.java
//package no.nav.sosialhjelp.soknad.consumer.kodeverk.dto;
//
//import com.fasterxml.jackson.core.JsonProcessingException;
//import com.fasterxml.jackson.databind.ObjectMapper;
//import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
//import org.junit.jupiter.api.Test;
//
//import static org.assertj.core.api.Assertions.assertThat;
//
//class KodeverkDtoTest {
//
//    @Test
//    void skalDeserialisereResponse() throws JsonProcessingException {
//        String json = "{\"betydninger\":{ \"NOR\": [ {\"gyldigFra\": \"1900-01-01\",\"gyldigTil\": \"9999-12-31\",\"beskrivelser\": {\"nb\": {\"term\": \"NORGE\",\"tekst\": \"NORGE\" } } }] } } ";
//
//        KodeverkDto response = new ObjectMapper()
//                .registerModule(new JavaTimeModule())
//                .readerFor(KodeverkDto.class)
//                .readValue(json);
//
//        assertThat(response).isNotNull();
//        assertThat(response.getBetydninger().get("NOR").get(0).getBeskrivelser().get("nb").getTerm()).isEqualTo("NORGE");
//    }
//}

File: src/test/kotlin/no/nav/sosialhjelp/soknad/client/kodeverk/dto/KodeverkDtoTest.kt
package no.nav.sosialhjelp.soknad.client.kodeverk.dto

import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import com.fasterxml.jackson.module.kotlin.readValue
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class KodeverkDtoTest {

    @Test
    internal fun `skal deserialisere response`() {
        val json = """{"betydninger":{ "NOR": [ {"gyldigFra": "1900-01-01","gyldigTil": "9999-12-31","beskrivelser": {"nb": {"term": "NORGE","tekst": "NORGE" } } }] } } """

        val response = jacksonObjectMapper()
            .registerModule(JavaTimeModule())
            .readValue<KodeverkDto>(json)

        assertThat(response).isNotNull
        assertThat(response.betydninger!!["NOR"]!![0].beskrivelser!!["nb"]!!.term).isEqualTo("NORGE")
    }
}
